opensrp-chw-core/src/test/java/org/smartregister/chw/core/utils/VisitVaccineUtilTest.java

```diff
@@ -21,69 +20,98 @@
+
+        VaccineRepo.Vaccine childVaccine = VaccineRepo.Vaccine.measles1;
+        VaccineCache childVaccineCache = new VaccineCache();
+        childVaccineCache.vaccineRepo = Collections.singletonList(childVaccine);
+
+        when(vaccineMapMock.get(eq("woman"))).thenReturn(womanVaccineCache);
+        when(vaccineMapMock.get(eq("child"))).thenReturn(childVaccineCache);
+
         when(ImmunizationLibrary.getVaccineCacheMap())
                 .thenReturn(vaccineMapMock);
 
-        VisitVaccineUtil.getAllVaccines();
-        verify(vaccineMapMock).get(eq("woman"));
-        verify(vaccineMapMock).get(eq("child"));
+        Map<String, VaccineRepo.Vaccine> vaccinesMap = VisitVaccineUtil.getAllVaccines();
+        Assert.assertTrue(vaccinesMap.containsValue(womanVaccine));
+        Assert.assertTrue(vaccinesMap.containsValue(childVaccine));
     }
 
     @Test
     public void testWrapVaccine(){
         VaccineRepo.Vaccine vaccine = VaccineRepo.Vaccine.bcg;
         Alert alert = mock(Alert.class);
         VaccineWrapper wrapped = VisitVaccineUtil.wrapVaccine(vaccine, alert);
-        Assert.assertEquals(wrapped.getVaccine(), vaccine);
-        Assert.assertEquals(wrapped.getName(), vaccine.display());
-        Assert.assertEquals(wrapped.getDefaultName(), vaccine.display());
-        Assert.assertEquals(wrapped.getAlert(), alert);
+        Assert.assertEquals(vaccine, wrapped.getVaccine());
+        Assert.assertEquals(vaccine.display(), wrapped.getName());
+        Assert.assertEquals(vaccine.display(), wrapped.getDefaultName());
+        Assert.assertEquals(alert, wrapped.getAlert());
     }
 
     @Test
     public void testGroupByTypeReturnsGrouped(){
-        Alert alert = spy(new Alert("case id", "case scheduled name", "visit code", AlertStatus.upcoming, "startDate", "expiryDate"));
-        String type = "case scheduled nam";
-        List<Alert> alerts  = Collections.singletonList(alert);
+        String scheduleName1 = "Ante Natal Care - Normal";
+        String scheduleName2 = "TT";
+        Alert scheduleName1Alert1 = new Alert("Case X", scheduleName1, "ANC 1", AlertStatus.normal, "2012-01-01", "2012-01-11");
+        Alert scheduleName1Alert2 = new Alert("Case Y", scheduleName1, "ANC 2", AlertStatus.complete, "2012-01-01", "2012-01-11");
+        Alert scheduleName2Alert = new Alert("Case X", scheduleName2, "TT 1", normal, "2012-01-01", "2012-01-11");
+        List<Alert> alerts  = Arrays.asList(scheduleName1Alert1, scheduleName1Alert2, scheduleName2Alert);
+
         Map<String, List<Alert>> grouped = VisitVaccineUtil.groupByType(alerts);
-        Assert.assertTrue(grouped.containsKey(type));
+
+        String type1Key = scheduleName1.substring(0, scheduleName1.length() - 1).trim();
+        String type2Key = scheduleName2.substring(0, scheduleName2.length() - 1).trim();
+        Assert.assertTrue(grouped.get(type1Key).containsAll(Arrays.asList(scheduleName1Alert1, scheduleName1Alert2)));
```

Comment:
Probably check the sizes here. This makes it a bit more "definite", for lack of a better word.